### PR TITLE
Set default Kafka command consumer assignment strategy

### PR DESCRIPTION
**Pull request by calohmn**
Using ~[StickyAssignor](https://kafka.apache.org/27/javadoc/org/apache/kafka/clients/consumer/StickyAssignor.html)~ [CooperativeStickyAssignor](https://kafka.apache.org/27/javadoc/org/apache/kafka/clients/consumer/CooperativeStickyAssignor.html) as default partition assignment strategy for the command router Kafka consumer.

Compared to the default assignor (RangeAssignor) this produces a more balanced and sticky assignment.

### CODE REVIEW

**Comment by calohmn** [comment]
I've now addressed the above issue and also added further improvements/fixes concerning the rebalance handling.
3 test runs, running just the Kafka C&C tests, were green (2 complete test runs showed the #2916 error - I'm currently working on that).
@b-abel Could you have a look again?

**Comment by calohmn** [comment]
Rebased, so that fix for #2916 is included.

**Comment by calohmn** [comment]
@b-abel I've updated the commits and improved the fix regarding https://github.com/eclipse/hono/pull/2908#issuecomment-950906125.

**Comment by b-abel** [comment]
@calohmn Won't #2925 make significant portions of this obsolete?

**Comment by calohmn** [comment]
@b-abel It will indeed have the effect, that all the logic behind `isAutoOffsetResetConfigLatest()` checks will not be used in the context of the Kafka command consumer.
But for the sake of consistency and because `HonoKafkaConsumer` is kept generic, I would keep the changes here.

In the PR for #2925, we can see, whether code can be removed.

File: clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AsyncHandlingAutoCommitKafkaConsumer.java [hunk 1]
```diff
@@ -425,36 +463,41 @@ private void commitOffsetsSync() {
...
+                        if (error != null) {
+                            log.info("periodic commit failed: {} [client-id: {}]", error, getClientId());
+                        } else {
+                            log.trace("periodic commit succeeded");
+                            setCommittedOffsets(committedOffsets);
+                        }
+                    });
+                } catch (final RebalanceInProgressException ex) {
+                    log.info("could not do periodic commit [client-id: {}]", getClientId(), ex);
+                    if (isCooperativeRebalancingConfigured()) {
```
[the remark this answers is not in the document]
Comment (calohmn) [reply clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AsyncHandlingAutoCommitKafkaConsumer.java:490]: Right, I've changed it to DEBUG here.

**Comment by b-abel** [comment]
LGTM

**Pull request merged**